src/fastoad/models/performances/mission.py

```diff
@@ -0,0 +1,252 @@
+        self.add_input("data:geometry:aircraft:wing:area", np.nan, units="m**2")
+
+        # TODO: solver needs an initial guess, how to not hard code it ?
+        self.add_output(
+            "data:mission:sizing:cruise:altitude",
+            0.3048 * np.full(shape, 35000),
+            shape=shape,
+            units="m",
+        )
+
+        self.declare_partials("data:mission:sizing:cruise:altitude", "*", method="fd")
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        altitude_interpolation = self.options["altitude_interpolation"]
+        initial_altitude = inputs["data:mission:sizing:cruise:altitude:initial"]
+        speed = inputs["data:mission:sizing:cruise:speed"]
+        weight = inputs["data:mission:sizing:cruise:weight"]
+        optimum_cl = inputs["data:aerodynamics:aircraft:cruise:optimal_CL"]
+        wing_area = inputs["data:geometry:aircraft:wing:area"]
+
+        rho = 2 * weight * g / optimum_cl / wing_area / speed ** 2
+
+        altitude = altitude_interpolation(rho)
+        altitude[0] = initial_altitude
+
+        outputs["data:mission:sizing:cruise:altitude"] = altitude
+
+
+class _CruiseThrust(om.ExplicitComponent):
+    """
+    Estimation of thrust vector
+    """
+
+    def initialize(self):
+        self.options.declare("flight_point_count", 50, types=(int, tuple))
+
+    def setup(self):
+
+        shape = self.options["flight_point_count"]
+        self.add_input("data:mission:sizing:cruise:altitude", np.nan, shape=shape, units="m")
+        self.add_input("data:TLAR:cruise_mach", np.nan, shape=shape)
+        self.add_input("data:mission:sizing:cruise:weight", np.nan, shape=shape, units="kg")
+        self.add_input("data:aerodynamics:aircraft:cruise:optimal_CL", np.nan)
+        self.add_input("data:aerodynamics:aircraft:cruise:optimal_CD", np.nan)
+
+        self.add_output("data:propulsion:phase", FlightPhase.CRUISE, shape=shape)
+        self.add_output("data:propulsion:use_thrust_rate", False, shape=shape)
+        self.add_output(
+            "data:propulsion:required_thrust_rate", 0.0, shape=shape, lower=0.0, upper=1.0
+        )
+        self.add_output("data:propulsion:required_thrust", shape=shape, units="N")
+        self.add_output("data:propulsion:altitude", shape=shape, units="m")
+        self.add_output("data:propulsion:mach", shape=shape)
+
+        self.declare_partials("data:propulsion:phase", "*", method="fd")
+        self.declare_partials("data:propulsion:use_thrust_rate", "*", method="fd")
+        self.declare_partials("data:propulsion:required_thrust_rate", "*", method="fd")
+        self.declare_partials("data:propulsion:required_thrust", "*", method="fd")
+        self.declare_partials("data:propulsion:altitude", "*", method="fd")
+        self.declare_partials("data:propulsion:mach", "*", method="fd")
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/fast-aircraft-design/FAST-OAD/pullRequest?prid=5618066)